File: frontend/src/types/protein.ts
export interface ProteinSearchResult {
  accession: string;
  name: string;
  organism: string;
  length: number;
  source: "ncbi" | "uniprot";
}

export interface ProteinDetail {
  accession: string;
  name: string;
  full_name?: string;
  organism: string;
  tax_id?: number;
  length: number;
  function?: string;
  gene_name?: string;
  source: string;
}

export interface ProteinSequence {
  accession: string;
  sequence: string;
  length: number;
  source: string;
}

File: backend/app/schemas/protein.py
from pydantic import BaseModel


class ProteinSearchQuery(BaseModel):
    query: str
    organism: str | None = None
    limit: int = 20


class ProteinSearchResult(BaseModel):
    accession: str
    name: str
    organism: str
    length: int
    source: str  # "ncbi" or "uniprot"


class ProteinDetail(BaseModel):
    accession: str
    name: str
    full_name: str | None = None
    organism: str
    tax_id: int | None = None
    length: int
    function: str | None = None
    gene_name: str | None = None
    source: str


class ProteinSequence(BaseModel):
    accession: str
    sequence: str
    length: int
    source: str
